### ISSUE TITLE
Blank state transitions do not update to state data

### ISSUE THREAD

**Issue by nicbus** [RGB-WG/rgb-wallet#80]
Using our rgb-sandbox I tried to issue 2 assets to the same UTXO, then transfer the first asset to another wallet and then check the the second asset's allocation, which I expect to be on the change outpout at that point.

What I find is that, after the transfer has completed, the second asset's allocation is still on the original issuance UTXO, which has now been spent.

To give an example, this is the situation after issuing the two assets:
- wallet unspents:
 ```
    8a675edd1a228bdbe24254bf1675a057275b49ade7bac9049513a77a40c6f59b:0
```
- 1st asset allocations
```
    amount=2000  utxo=8a675edd1a228bdbe24254bf1675a057275b49ade7bac9049513a77a40c6f59b:0
```
- 2nd asset allocations
```
    amount=2000  utxo=8a675edd1a228bdbe24254bf1675a057275b49ade7bac9049513a77a40c6f59b:0
```

Then a transfer (txid `296ab67bd423551c87229e9371b85ee5c62b60417d8015bed146d49d71bb784f`) sends 100 units of the first asset to another wallet and this is the resulting situation (on the issuer/sender side):
- wallet unspents:
 ```
    296ab67bd423551c87229e9371b85ee5c62b60417d8015bed146d49d71bb784f:1
```
- 1st asset allocations
```
    amount=1900  utxo=296ab67bd423551c87229e9371b85ee5c62b60417d8015bed146d49d71bb784f:1
```
- 2nd asset allocations
```
    amount=2000  utxo=8a675edd1a228bdbe24254bf1675a057275b49ade7bac9049513a77a40c6f59b:0
```

This issue can be reproduced using [this branch](https://github.com/nicbus/rgb-sandbox/tree/v0.10_blank) of rgb-sandbox.

**Comment by dr-orlovsky** [comment]
I was able to replicate the issue, thank you for the sandbox environment.

From my debug information I see that the blank transition is created and the state is transferred correctly, i.e. stash state is correct and nothing is lost. The problem is that the blank transition is not indexed and not parsed to update cached state data in the inventory, so rgb can't "see" it.

Investigating the reasons why it happens.

**Comment by dr-orlovsky** [comment]
Was this working correctly before the latest PRs?

**Comment by nicbus** [comment]
> Was this working correctly before the latest PRs?

Unfortunately I can't tell.

Checking a blank transfer was not part of rgb-sandbox before, as we were focusing on more basic transfer functionality, and has just been added in the attempt to cover more and more use cases.

**Comment by crisdut** [comment]
This error is the same of that #60

**Comment by dr-orlovsky** [comment]
Fixed with #81.

The tests with sandbox now pass for me (I updated the `rgb-contracts` crate to use the fixed version of `rgb-std`)
